### PR TITLE
Add ocean surface mass flux climatology task

### PR DESCRIPTION
**Pull request by cbegeman**
This PR adds a climatology task for ocean surface mass fluxes. It also fixes a small issue with raising errors for adding tasks.

### CODE REVIEW

**Comment by cbegeman** [comment]
### Testing

This analysis has been run on chrysalis including comparison against a control run: https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.cbegeman/analysis/20240611.HRr5_test9.chrysalis/yrs1-10_clim1-10/ocean/index.html

**Review by xylar — COMMENTED** [review]
I'm trying to test this on Chrysalis but even single-node jobs are waiting for hours in the queue.

In the meantime, let's discuss the `failureCount` changes below.

File: mpas_analysis/__main__.py [hunk 1]
```diff
@@ -433,11 +436,11 @@ def add_task_and_subtasks(analysisTask, analysesToGenerate, verbose,
                                              verbose, callCheckGenerate=False)
         totalFailures += failureCount
         if prereq._setupStatus != 'success':
-            assert(failureCount > 0)
-            # a prereq failed setup_and_check
             print("Warning: prerequisite of {} failed during check, "
                   "so this task will not be run".format(
                       taskTitle))
+            assert(failureCount > 0)
+            # a prereq failed setup_and_check
```
[the remark this answers is not in the document]
Comment (xylar) [reply mpas_analysis/__main__.py:443]: Maybe something like:
``` python
            if failureCount == 0:
                raise ValueError(f'prerequisite {prereq.printTaskName} of {taskTitle} did not set up successfully but also '
                                 f'did not indicate a failure.  This likely indicates a bug like multiple tasks with the same name.')
            # a prereq failed setup_and_check
            print("Warning: prerequisite of {} failed during check, "
                  "so this task will not be run".format(
                      taskTitle))
```
(Note, formatting likely will need to be fixed.)

**Comment by xylar** [comment]
@cbegeman, this is really great!  Other than the issue above that we can chat about, I think this is ready to go.  I ran the test suite on Chrysalis and results are here:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xasay-davis/analysis_testing/chrysalis/ocn-add-surface-flux-climo/

File: mpas_analysis/__main__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbegeman) [reply mpas_analysis/__main__.py]: @xylar Thanks for your suggestion. I have made the changes and run again but I did not try to trigger the error. https://github.com/MPAS-Dev/MPAS-Analysis/pull/1007/commits/137d561b9cac89dd46c1deedbde565314a1ecb7f

[the remark this answers is not in the document]
Comment (xylar) [reply mpas_analysis/__main__.py]: Yep, that's fine.  I don't think you need to worry about triggering the error again.

**Comment by xylar** [comment]
Thanks for fixing those error messages!

I think the only remaining thing would be to add some documentation for the task.

I think the CI failure is a temporary one unrelated to this branch.

**Comment by cbegeman** [comment]
@xylar Are you happy with this analysis being included in the default config as it is now?

**Comment by xylar** [comment]
@cbegeman, yes, I'm okay with that.  I think it will be useful for all runs.  I think the Arctic/Antarctic focus is appropriate even for simulations that aren't otherwise focused on polar regions.

**Comment by cbegeman** [comment]
@xylar Great! Thanks for reviewing. Would you like to take a look at the docs before I merge? I'm writing and checking the html this afternoon

**Review by xylar — APPROVED** [review]

**Comment by xylar** [comment]
I'm happy to trust you on the docs.  Feel free to merge once they're in.

**Pull request merged**